### ISSUE TITLE
Builds run after UNSTABLE build are stuck

### ISSUE THREAD

**Issue by d2lam** [screwdriver-cd/screwdriver#1152]
**What happened**:
In current build `update` logic, we check if `status` field is `dirty`: https://github.com/screwdriver-cd/models/blob/master/lib/build.js#L263-L274
However, because `UNSTABLE` cannot be changed to something else (`SUCCESS` or `FAILURE`) afterward, when the build finishes, the field is not `dirty`. As a result, `stop` will never get called and `running_key` will not get cleaned up. 

**How to reproduce it**:
- Run a build and update it to `UNSTABLE`
- Subsequent builds are blocked

<img width="516" alt="screen shot 2018-06-26 at 12 11 00 pm" src="https://user-images.githubusercontent.com/3401924/41933907-057758ba-793a-11e8-87f7-00770834b415.png">

**Comment by d2lam** [comment]
I looked into this issue and it is a little complicated since `UNSTABLE` case is too special.

Currently it uses the check if `status` is dirty, and if the build is `done` (`ABORTED`, `SUCCESS`, `FAILURE`) then it will call `stop`

However, both these cases don't apply for `UNSTABLE`. If we defined `UNSTABLE` as `done` then it is also wrong because according to https://github.com/screwdriver-cd/screwdriver/issues/1122, the build is not done and should still proceed to the next steps.
